heka/sandbox/outputs/redshift.lua

```diff
@@ -0,0 +1,155 @@
+
+--[[
+Example Redshift database message loader. The plugin includes table setup,
+message transformation and file buffering to improve performance by bulk
+loading data.  The bulk load will be triggered by a full buffer or by the
+flush_interval.
+
+The output will fail to start if the Redshift cluster is unavailable.  If the
+Redshift cluster becomes unavailable while this plugin is running it will
+backpressure Heka causing all data flowing through Heka to pause.  Ideally
+all output plugins should be run in a separate instance of Heka and read from a
+persistent store (file, Kafka, etc) to avoid impacting any core monitoring
+operations.
+
+Config:
+
+- db_name (string)
+- db_user (string)
+- db_password (string)
+- db_host (string)
+- db_port (uint)
+- buffer_file (string)
+    Path to the buffer file
+- buffer_size (uint, optional, default 10240000)
+    File size, in bytes, of table inserts to buffer before bulk loading.
+- flush_interval (uint, optional, default 60)
+    The maximum amount of time in seconds between bulk load operations.  This
+    should match the plugin ticker_interval configuration.
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [RedshiftOutput]
+    type = "SandboxOutput"
+    filename = "redshift.lua"
+    message_matcher = "Type == 'logfile'"
+    memory_limit = 60000000
+    ticker_interval = 60
+
+        [RedshiftOutput.config]
+        db_name = "dev"
+        db_user = "testuser"
+        db_password = "testuserpw"
+        db_host = "foobar.redshift.amazonaws.com"
+        db_port = 5439
+        buffer_file = "/var/tmp/redshift.insert"
+        flush_interval = 60
+--]]
+
+require "io"
+require "os"
+require "string"
+
+local driver = require "luasql.postgres"
+
+-- db setup
+local db_name     = read_config("db_name") or error("db_name must be set")
+local db_user     = read_config("db_user") or error("db_user must be set")
+local db_password = read_config("db_password") or error("db_password must be set")
+local db_host     = read_config("db_host") or error("db_host must be set")
+local db_port     = read_config("db_port") or error("db_port must be set")
+local buffer_file = read_config("buffer_file") or error("buffer_file must be set")
+
+local buffer_size = read_config("buffer_size") or 10000 * 1024
+assert(buffer_size > 0, "buffer_size must be greater than zero")
+
+local flush_interval = read_config("flush_interval") or 60
+assert(flush_interval >= 0, "flush_interval must be greater than or equal to zero")
+flush_interval = flush_interval * 1e9
+
+local env = assert (driver.postgres())
+local con, err = env:connect(db_name, db_user, db_password, db_host, db_port)
+assert(con, err)
+assert (con:execute[[
+        CREATE TABLE IF NOT EXISTS logs(
+        Timestamp bigint,
+        Severity integer,
+        Pid integer,
+        Hostname varchar(126),
+        Type varchar(126),
+        Logger varchar(126),
+        EnvVersion varchar(126),
+        Payload varchar(126)
```

Comment:
Postgres will fold these column names to lowercase - if you want them in mixed-case, they should be enclosed in double-quotes.  If you do not want them in mixed-case, you should use the lowercase names in the CREATE TABLE statement.  And as a style nit, I like to see the SQL data types in uppercase (`VARCHAR` vs. `varchar`).